api/app.js

```diff
@@ -143,9 +151,27 @@ if (cluster.isMaster){
   //Allow cross origin requests
   app.use(cors());
   app.use(compression()); //Compress all routes
+
+  // auth function for the hookswe
+  hookAuth = (req, res, next) => {
+    const b64auth = (req.headers.authorization || '').split(' ')[1] || ''
+    const strauth = new Buffer(b64auth, 'base64').toString()
+    const splitIndex = strauth.indexOf(':')
+    const user = strauth.substring(0, splitIndex)
+    const password = strauth.substring(splitIndex + 1)
+    if (user && password && user === "webhookuser" && password === process.env.WEBHOOK_PASS) {
+      // grant access
+      return next()
+    }
+    // access denied
+    res.set('WWW-Authenticate', 'Basic realm="401"')
```

Comment:
I think it might be a good idea to not require auth if this `WEBHOOK_PATH` is missing or not set in the env so they can continue to use these without auth if they so choose.

Reply:
Changed it to pass through if `WEBHOOK_PASS` is `null` or `""`.